From: Ben Hutchings <ben@decadent.org.uk>
Date: Tue, 10 Sep 2013 03:32:23 +0100
Subject: mm: Avoid ABI change in 3.2.51

Move the added field mm_struct::mmap_legacy_base to the end of the
struct and hide it from genksyms.

--- a/include/linux/mm_types.h
+++ b/include/linux/mm_types.h
@@ -297,7 +297,6 @@ struct mm_struct {
 	void (*unmap_area) (struct mm_struct *mm, unsigned long addr);
 #endif
 	unsigned long mmap_base;		/* base of mmap area */
-	unsigned long mmap_legacy_base;         /* base of mmap area in bottom-up allocations */
 	unsigned long task_size;		/* size of task vm space */
 	unsigned long cached_hole_size; 	/* if non-zero, the largest hole below free_area_cache */
 	unsigned long free_area_cache;		/* first hole of size cached_hole_size or larger */
@@ -390,6 +389,10 @@ struct mm_struct {
 #ifdef CONFIG_CPUMASK_OFFSTACK
 	struct cpumask cpumask_allocation;
 #endif
+
+#ifndef __GENKSYMS__
+	unsigned long mmap_legacy_base;         /* base of mmap area in bottom-up allocations */
+#endif
 };
 
 static inline void mm_init_cpumask(struct mm_struct *mm)
